src/Controller/HearingController.php

```diff
@@ -65,11 +66,17 @@ public function index(CaseEntity $case, HearingPostRepository $hearingPostReposi
             $this->entityManager->flush();
         }
 
-        $partyHasSomethingToAdd = $hasCounterparty
-            ? true === $hearing->getPartyHasNoMoreToAdd() && true === $hearing->getCounterpartHasNoMoreToAdd()
-            : true === $hearing->getPartyHasNoMoreToAdd();
+        $partyHasSomethingToAdd = false;
 
-        $form = $this->createForm(HearingFinishType::class, $hearing, ['case' => $case, 'hasCounterparty' => $hasCounterparty]);
+        if ($hasCounterparty && $hasParty) {
+            $partyHasSomethingToAdd = !($hearing->getCounterpartHasNoMoreToAdd() && $hearing->getPartyHasNoMoreToAdd());
+        } elseif ($hasCounterparty) {
+            $partyHasSomethingToAdd = !$hearing->getCounterpartHasNoMoreToAdd();
+        } elseif ($hasParty) {
+            $partyHasSomethingToAdd = !$hearing->getPartyHasNoMoreToAdd();
+        }
```

Comment:
I think this can be simply written as

```php
$partyHasSomethingToAdd = ($hasCounterparty && !$hearing->getCounterpartHasNoMoreToAdd()) 
  || ($hasParty && !$hearing->getPartyHasNoMoreToAdd()).
```

Reply:
It can indeed, nice.